cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/proto/summary/ReportSummary.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *  
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *  
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.report.proto.summary;
+
+import com.google.common.base.Objects;
+
+import java.util.Set;
+
+/**
+ * Represents the summary of a program operation status report in an HTTP response.
+ */
+public class ReportSummary {
+  private final Set<NamespaceAggregate> namespaces;
+  private final long start;
+  private final long end;
+  private final Set<ArtifactAggregate> artifacts;
+  private final DurationStats durations;
+  private final StartStats starts;
+  private final Set<UserAggregate> owners;
+  private final Set<StartMethodAggregate> startMethods;
+
+  public ReportSummary(Set<NamespaceAggregate> namespaces, long start, long end,
+                       Set<ArtifactAggregate> artifacts,
+                       DurationStats durations, StartStats starts, Set<UserAggregate> owners,
+                       Set<StartMethodAggregate> startMethods) {
+    this.namespaces = namespaces;
+    this.start = start;
+    this.end = end;
+    this.artifacts = artifacts;
+    this.durations = durations;
+    this.starts = starts;
+    this.owners = owners;
+    this.startMethods = startMethods;
+  }
+
+  /**
+   * @return the number of program runs in each unique namespaces
+   */
+  public Set<NamespaceAggregate> getNamespaces() {
+    return namespaces;
+  }
+
+  /**
+   * @return the start of the time range within which the report is generated. All program runs in the report must be
+   *         still running at {@code start}, or with end time not earlier than {@code start}.
+   */
+  public Long getStart() {
+    return start;
+  }
+
+  /**
+   * @return the end of the time range within which the report is generated. All program runs in the report must
+   *         start before {@code end}.
+   */
+  public Long getEnd() {
+    return end;
+  }
+
+  /**
+   * @return the number of program runs in each unique parent artifact
+   */
+  public Set<ArtifactAggregate> getArtifacts() {
```

Comment:
why these return types are set ? since you are aggregating, runs belonging to artifact/namespace, etc will be added up in corresponding object, not sure why you need a set.

Reply:
this is to help comparing whether two ReportSummary's are the equal. They should be equal regardless of the order of artifacts.

Reply:
is it really needed ? you create report summary object in scala and write it to a file and you read and deserialize and return.

Reply:
right, but this helps the unit test, making it a list will complicates the `equals` method, and it doesn't have much additional cost